pkg/controller/volume/attachdetach/reconciler/reconciler.go

```diff
@@ -182,8 +182,15 @@ func (rc *reconciler) reconcile() {
 
 	// Ensure volumes that should be detached are detached.
 	for _, attachedVolume := range rc.actualStateOfWorld.GetAttachedVolumes() {
-		if !rc.desiredStateOfWorld.VolumeExists(
-			attachedVolume.VolumeName, attachedVolume.NodeName) {
+
+		shouldStartOperation := !rc.desiredStateOfWorld.VolumeExists(attachedVolume.VolumeName, attachedVolume.NodeName)
+
+		if utilfeature.DefaultFeatureGate.Enabled(features.NodeShutdown) {
```

Comment:
if we are doing this with featuregate in this point, it means that we have then bug in whole attach+detach thing. 

Lets think following scenario:
- we have lets say aws cloudprovider and node is shutdown (node is not deleted anymore from cluster)
- featuregate not enabled

the result is that volume does not detach from node which is shutdown. IMO remove featuregate from this one because cloudprovider changes is done already (like in openstack it has been for a long time already).